lib/storage/src/content_manager/consensus_state.rs

```diff
@@ -227,6 +263,7 @@ impl<C: CollectionContainer> ConsensusState<C> {
             };
             let do_increase_applied_index: bool = if entry.data.is_empty() {
                 // Empty entry, when the peer becomes Leader it will send an empty entry.
+                self.is_leader_established.make_ready();
```

Comment:
Why is this a better location to set `make_ready()`?

Reply:
I tried to remove all sleeps from the consensus test. There we assert that we have this empty entry when the leader was elected, and as you can notice it happens a few lines after the conf change.

So this just gives us stricter guarantees.